hardware/national_instruments_x_series.py

```diff
@@ -1495,49 +1506,67 @@ def set_up_odmr(self, counter_channel=None, photon_source=None,
         else:
             my_clock_channel = self._scanner_clock_channel
 
-        if counter_channel is not None:
-            my_counter_channel = counter_channel
-        else:
-            my_counter_channel = self._scanner_counter_channels[0]
+        if len(self._scanner_counter_channels) > 0 and len(self._photon_sources) > 0:
```

Comment:
if you want to check against empty lists (or tuples, dicts etc.) you can also just ask `if first_list and second_list:`.